aderyn_core/src/report/markdown_printer.rs

```diff
@@ -0,0 +1,231 @@
+                    w,
+                    "  - [{}-{}: {}](#{}-{}-{})",
+                    severity,
+                    index + 1,
+                    issue.title,
+                    severity.to_ascii_lowercase(),
+                    index + 1,
+                    issue_title_slug
+                )
+                .unwrap();
+            }
+        }
+
+        fn display<T: Write>(title: &str, issues: &[Issue], mut writer: T, severity: &str) {
+            if !issues.is_empty() {
+                writeln!(writer, "{}", title).unwrap();
+                print_table_of_content(issues, &mut writer, severity);
+            }
+        }
+
+        writeln!(writer, "# Table of Contents\n")?;
+        writeln!(writer, "- [Summary](#summary)")?;
+        writeln!(writer, "  - [Files Summary](#files-summary)")?;
+        writeln!(writer, "  - [Files Details](#files-details)")?;
+        writeln!(writer, "  - [Issue Summary](#issue-summary)")?;
+
+        let issues = [
+            (
+                &report.criticals,
+                "- [Critical Issues](#critical-issues)",
+                "C",
+            ),
+            (&report.highs, "- [High Issues](#high-issues)", "H"),
+            (&report.mediums, "- [Medium Issues](#medium-issues)", "M"),
+            (&report.lows, "- [Low Issues](#low-issues)", "L"),
+            (&report.ncs, "- [NC Issues](#nc-issues)", "NC"),
+        ];
+
+        issues.iter().for_each(|rec| {
+            display(rec.1, rec.0, &mut writer, rec.2);
+        });
+
+        writeln!(writer, "\n")?; // Add an extra newline for spacing
+        Ok(())
+    }
+
+    fn print_issue<W: Write>(
+        &self,
+        mut writer: W,
+        issue_body: &IssueBody,
+        severity: &str,
+        number: i32,
+        root_path: &Path,
+    ) -> Result<()> {
+        let is_file = root_path.is_file();
+
+        writeln!(
+            writer,
+            "## {}-{}: {}\n\n{}\n", // <a name> is the anchor for the issue title
+            severity, number, issue_body.title, issue_body.description
+        )?;
+        for instance in &issue_body.instances {
+            let path = {
+                if is_file {
+                    String::from(root_path.to_str().unwrap())
+                } else {
+                    String::from(
+                        root_path
+                            .join(instance.contract_path.clone())
+                            .as_path()
+                            .to_str()
+                            .unwrap(),
+                    )
+                }
+            };
+
+            let line = std::fs::read_to_string(&path).unwrap();
+
+            let line_preview = line
+                .split('\n')
+                .skip(instance.line_no - 1)
+                .take(1)
+                .next()
+                .unwrap();
+
+            writeln!(
+                writer,
+                "- Found in {} Line: {}\n\n\t```solidity\n\t{}\n\t```\nfile://{}\n\n\n\n",
```

Comment:
Could we append the line number in the `file://` so that on ctrl+click it opens the file on the correct line?

Reply:
Also, if we remove the line_preview, we should make the `file://` url a markdown link. As in:

```md
- Found in [Line: 35](file://path/to/file.sol:35)
```

Reply:
Hi Alex! I have changed it a bit. Please can you check again and see if you like it? The links are now clickable from within the markdown-rendered view.  However, I didn't remove the line preview yet : ) 

So now, that we're not printing `file:///blah/blah/blah` it conserves some space. And if you still think line preview isn't going to add too much value I'll remove it

Reply:
This looks great!